.github/workflows/docs-preview-trigger.yml

```diff
@@ -0,0 +1,195 @@
+name: Trigger Docs Preview Build
+
+on:
+    pull_request:
+        paths:
+            - 'docs/published/**'
+    push:
+        branches:
+            - master
+        paths:
+            - 'docs/published/**'
+
+permissions:
+    pull-requests: write
+
+jobs:
+    trigger-preview:
+        name: Trigger posthog.com preview build
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+
+            - name: Trigger Vercel build with inline env
+              id: trigger-build
+              env:
+                  VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
+                  VERCEL_TEAM_ID: ${{ secrets.VERCEL_TEAM_ID }}
+                  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+              run: |
```

Comment:
Maybe pull this out to a script? It's a bit messy in the workflow file directly because of the giant indent